src/cleo/commands/base_command.py

```diff
@@ -114,12 +114,7 @@ def run(self, io: IO) -> int:
 
         io.input.validate()
 
-        status_code = self.execute(io)
-
-        if status_code is None:
-            status_code = 0
-
-        return status_code
+        return self.execute(io)
```

Comment:
`self.execute` returns an `int` (or at least that's what the type hint says)

Reply:
Yes, `self.execute()` returns an `int`, never `None`. I didn't find any implementation to disobey this.

Reply:
yes, but the most commonly used implementation of `execute()` is found here:
https://github.com/python-poetry/cleo/blob/016e0d2568bb9f16a7d22505428982e80aa3214b/src/cleo/commands/command.py#L57-L63

and since the `handle()` method is the one being overwritten by the library user, we should prevent accidental breakage and send `None` up the chain (which would end up as a status code returned by the application). Of course, in a perfect world users would always make their `handle()` methods correct, but let's assume that they might forget to return the status code :wink:

Reply:
Not to mention that this would be considered a breaking change, since the default behaviour would change.

Reply:
Then `execute` should be changed instead to make sure it always returns an `int`

Reply:
How about correcting the type hint of `self.handle()`, allowing `None` as the return value, to allow it explicitly, since that might be more convenient?

Reply:
I would prefer if we could just put the `or 0` at the return for now and get back to this discussion when we are refactoring the code for Cleo 3.0. Right now I would want to avoid breaking the behavior (even if it's not perfect).

Reply:
Done.